examples/geckocamera_example.cpp

```diff
@@ -50,29 +52,31 @@ class GeckoCameraExample : CameraListener
 
         cout << cameraList.size() << " cameras found\n";
 
-        if (cameraList.size()) {
-            CameraInfo info = cameraList.at(0);
+        if (cameraNumber < cameraList.size()) {
+            CameraInfo info = cameraList.at(cameraNumber);
             vector<CameraCapability> caps;
-            if (manager->queryCapabilities(info.id, caps)) {
+            if (manager->queryCapabilities(info.id, caps) && modeNumber < caps.size()) {
                 cout << "Camera " << info.id << " caps:\n";
                 for (auto cap : caps) {
                     cout << "    " << cap.width << "x" << cap.height << ":" << cap.fps << "\n";
                 }
 
                 shared_ptr<Camera> camera;
-                if (manager->openCamera(cameraList.at(0).id, camera)) {
+                if (manager->openCamera(cameraList.at(cameraNumber).id, camera)) {
```

Comment:
No need to do lookup again as you already have camera info variable at line 56. Sorry that I missed this before.